### PR TITLE
Enable linter and copyright tests to the package [7487]

### PR DESCRIPTION
**Pull request by Blast545**
Enable ament_common linter and copyright tests.
Added contributing.md file to pass these tests. 
Changed uppercase cmake command to lowercase to match linter configuration.

Signed-off-by: Jorge J. Perez <jjperez@ekumenlabs.com>

### CODE REVIEW

**Review by MiguelCompany — COMMENTED** [review]
I like the idea beneath this PR, but I don't like the idea of requiring new dependencies. Bear in mind that this package may be used outside ROS 2, and thus we don't want to impose those dependencies.

I suggest making the new dependencies optional and only use them when found.

**Comment by Blast545** [comment]
@MiguelCompany You are right, I added a commit addressing that. Dependencies are found using QUIET, and tests are run only when found.

Also specific tests are added, instead of the whole ament_cmake environment tools.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,9 @@ cmake_minimum_required(VERSION 3.5)
 project(foonathan_memory_vendor VERSION "0.5.0")
 
 find_package(foonathan_memory QUIET)
+find_package(ament_cmake_copyright QUIET)
+find_package(ament_cmake_lint_cmake QUIET)
+find_package(ament_cmake_xmllint QUIET)
```
Comment (hidmic) [inline CMakeLists.txt]: @Blast545 consider moving these into the `if (BUILD_TESTING) ... endif()` clause.

**Review by MiguelCompany — APPROVED** [review]
Looks good to me now

**Pull request merged**